driver-sync/src/test/functional/com/mongodb/client/model/expressions/InContextExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,160 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import com.mongodb.MongoClientSettings;
+import com.mongodb.client.AggregateIterable;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.model.Aggregates;
+import com.mongodb.client.model.Filters;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.mongodb.client.model.Accumulators.sum;
+import static com.mongodb.client.model.Aggregates.match;
+import static com.mongodb.client.model.Aggregates.project;
+import static com.mongodb.client.model.Projections.computed;
+import static com.mongodb.client.model.Projections.excludeId;
+import static com.mongodb.client.model.Projections.fields;
+import static com.mongodb.client.model.Projections.include;
+import static com.mongodb.client.model.Sorts.ascending;
+import static com.mongodb.client.model.expressions.Expressions.current;
+import static com.mongodb.client.model.expressions.Expressions.of;
+import static com.mongodb.client.model.expressions.Expressions.ofArray;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class InContextExpressionsFunctionalTest extends AbstractExpressionsFunctionalTest {
+
+    private MongoClient client;
+    private MongoCollection<Document> col;
+
+    @BeforeEach
+    public void setUp() {
+        client = MongoClients.create();
+        col = client.getDatabase("testdb").getCollection("testcol");
+        col.drop();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        client.close();
+    }
+
+    private String bsonToString(final Bson project) {
+        return project.toBsonDocument(Document.class, MongoClientSettings.getDefaultCodecRegistry()).toString().replaceAll("\"", "'");
+    }
+
+    private List<Document> aggregate(final Bson... stages) {
+        AggregateIterable<Document> result = col.aggregate(Arrays.asList(stages));
+        List<Document> results = new ArrayList<>();
+        result.forEach(r -> results.add(r));
+        return results;
+    }
+
+    @Test
+    public void matchTest() {
+        col.insertMany(Arrays.asList(
+                Document.parse("{_id: 1, x: 0, y: 2}"),
+                Document.parse("{_id: 2, x: 0, y: 3}"),
+                Document.parse("{_id: 3, x: 1, y: 3}")));
+
+        List<Document> results = aggregate(
+                match(Filters.expr(current().getInteger("x").eq(of(1)))));
```

Comment:
Was reading docs `match(expr(new Document("$and"`, seems that we might add a version of expr such that the above would become:

`Filters.expr(d -> d.getInteger("x").eq(of(1))`

Reply:
I am imagining that this could be something for the future.